banzai/images.py

```diff
@@ -131,6 +131,40 @@ def write_catalog(self, filename, nsources=None):
     def add_history(self, msg):
         self.header.add_history(msg)
 
+    def data_is_3d(self):
+        try:
+            data_is_3d = len(self.data.shape) > 2
+        except AttributeError:
+            logger.error("Trying to measure dimensions of data without a shape attribute",
+                          extra={'tags': {'filename': self.filename}})
+            data_is_3d = False
+        return data_is_3d
+
+    def get_n_amps(self):
+        if self.data_is_3d():
+            n_amps = self.data.shape[0]
+        else:
+            n_amps = 1
+        return n_amps
+
+    def get_inner_image_section(self, edge_fraction_denominator=4):
+        if self.data_is_3d():
+            logger.error("Cannot get inner section of a 3D image",
+                         extra={'tags': {'filename': self.filename}})
+            inner_section = self.data
+        else:
+            # Note that (-a//b) != -(a//b) if a is odd
+            try:
+                inner_nx = self.nx // edge_fraction_denominator
+                inner_ny = self.ny // edge_fraction_denominator
+                inner_section = self.data[inner_ny: -inner_ny, inner_nx: -inner_nx]
```

Comment:
I'm confused why this try is necessary?

Reply:
In case `nx`, `ny` or `data` are not set, then it will just return `data` as is and log an error, rather than crashing.